include/ada/url.h

```diff
@@ -132,7 +132,7 @@ namespace ada {
     /** Only for development purposes so we can see where the copies are happening. **/
     url oh_no_we_need_to_copy_url() const {
       url answer;
-      answer.scheme = scheme;
+      answer.non_special_scheme = non_special_scheme;
```

Comment:
Can we add a comment that `special_schemes` are stored in the `type` attribute, and why we made the decision to split it (performance) for future reference?

Reply:
Done.